vim/merlin/autoload/merlin.py

```diff
@@ -399,16 +400,18 @@ def vim_loclist(vimvar, ignore_warnings):
     bufnr = vim.current.buffer.number
     nr = 0
     for error in errors:
-        ty = 'E'
-        if error['type'] == 'warning':
-            if vim.eval(ignore_warnings) == 'true':
-                continue
-            ty = 'W'
```

Comment:
I don't understand that part of the diff, why did you delay that check? Aren't you just adding useless extra work?

Reply:
To prevent skipping over warn-error'd warnings when `ignore_warnings` is set.

Reply:
That part can then be reverted if the other change is implemented in the backend, as discussed.